core/trino-main/src/test/java/io/trino/operator/TestWorkProcessor.java

```diff
@@ -233,6 +233,33 @@ public void testYield()
         assertFinishes(processor);
     }
 
+    @Test(timeOut = 10_000)
+    public void testBlock()
+    {
+        SettableFuture<Void> baseFuture = SettableFuture.create();
```

Comment:
It might be easier to read if you gave `baseFuture` and `future` more consistent names. Maybe something like "phase1" and "phase2" or similar.